adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -137,6 +137,26 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
    */
   def replaceSequences(newSequences: SequenceDictionary): U
 
+  /**
+   * Caches underlying RDD in memory. This interfaced is used to
+   * access caching functionality from the python and R APIs.
+   *
+   * @return type of RDD that was cached
+   */
+  def cache() = {
+    rdd.cache()
```

Comment:
Asking out of ignorance, does caching make any sense for `GenomicDataset.dataset`?

Reply:
Ah, yes... These should probably be overridden by the `DatasetBound...RDD` classes.

Reply:
@fnothaft @heuermh I am confused what this means. Do you want me to implement this function in every DatasetBound...RDD function? There seems to be a lot of them, while this approach just requires 1 implementation.

Reply:
Hi @akmorrow13! Sorry for the slow reply here. Yes, that is what I want, but the simpler way to do this is to have `RDDBoundGenomicRDD` and `DatasetBoundGenomicRDD` traits that both implement the `cache`, `persist`, and `unpersist` methods. Then `RDDBoundAlignmentRecordRDD` would extend `RDDBoundGenomicRDD`, etc. We should do this for `saveAsParquet` as well.

Also, my preference is that `unpersist` returns `Unit` (no return value) and that `cache`/`persist` return either `U` or `Unit`.